pkg/scanners/types/types.go

```diff
@@ -43,7 +43,10 @@ type ImageScannerWithDataSource interface {
 // ImageVulnerabilityGetter is a scanner which can retrieve vulnerabilities
 // which exist in the given image components and the scan notes for the given image.
 type ImageVulnerabilityGetter interface {
-	GetVulnerabilities(image *storage.Image, components *scannerV1.Components, notes []scannerV1.Note) (*storage.ImageScan, error)
+	GetVulnerabilities(image *storage.Image, components any, notes []scannerV1.Note) (*storage.ImageScan, error)
```

Comment:
Same here I feel like this should be more strongly typed to avoid also the indirection of the newly introduce `CanHandle` method.

Reply:
See comment: https://github.com/stackrox/stackrox/pull/9342#discussion_r1450567359